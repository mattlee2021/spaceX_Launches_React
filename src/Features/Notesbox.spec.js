import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TestRenderer from "react-test-renderer";
import NotesBox from "./Notesbox";

Object.assign(navigator, {
  clipboard: {
    writeText: () => {},
  },
});

describe("NotesBox", () => {
  it("Should match the snapshot", () => {
    const testRenderer = TestRenderer.create(<NotesBox />);
    testRenderer.toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
  it("Should contain the user's input", () => {
    render(<NotesBox />);
    const notesbox = screen.getByRole("textbox");
    userEvent.type(notesbox, "Some Notes");
    expect(notesbox).toHaveValue("Some Notes");
  });

  it("Should copy the content of the notesbox to clipboard", () => {
    navigator.clipboard.writeText = jest.fn();
    render(<NotesBox />);
    const notesbox = screen.getByRole("textbox");
    userEvent.type(notesbox, "Some Notes");
    const copyToClipboardButton = screen.getByRole("button", {
      name: "Copy This Note To Clipboard",
    });
    userEvent.click(copyToClipboardButton);
    expect(navigator.clipboard.writeText).toHaveBeenCalledWith("Some Notes");
  });
});
