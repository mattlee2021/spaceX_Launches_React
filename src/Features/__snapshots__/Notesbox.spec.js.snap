// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesBox Should match the snapshot 1`] = `
<form
  className="notes_form"
>
  <div>
    <label>
      Notes
    </label>
    <br />
    <br />
    <textarea
      className="notes__input"
      onChange={[Function]}
      value=""
    />
  </div>
  <br />
  <button
    onClick={[Function]}
    type="button"
  >
    Copy This Note To Clipboard
  </button>
</form>
`;
